giddy/markov.py

```diff
@@ -547,6 +594,45 @@ class Spatial_Markov(object):
      [0.         0.         0.05660377 0.90566038 0.03773585]
      [0.         0.         0.         0.03932584 0.96067416]]
 
+    (3.1) As we can see from the above estimated conditional transition
+    probability matrix, some rows are full of zeros which violate the
+    requirement of a transition probability matrix that every row sums to 1.
+    We can easily adjust this assigning fill_diag = True when initializing
+    Spatial_Markov.
+    >>> sm = Spatial_Markov(rpci, w, cutoffs=cc, lag_cutoffs=cc, fill_diag=True)
+    >>> for p in sm.P:
+    ...     print(p)
+    [[0.96703297 0.03296703 0.         0.         0.        ]
+     [0.10638298 0.68085106 0.21276596 0.         0.        ]
+     [0.         0.14285714 0.7755102  0.08163265 0.        ]
+     [0.         0.         0.5        0.5        0.        ]
+     [0.         0.         0.         0.         1.        ]]
+    [[0.88636364 0.10606061 0.00757576 0.         0.        ]
+     [0.04402516 0.89308176 0.06289308 0.         0.        ]
+     [0.         0.05882353 0.8627451  0.07843137 0.        ]
+     [0.         0.         0.13846154 0.86153846 0.        ]
+     [0.         0.         0.         0.         1.        ]]
+    [[0.78082192 0.17808219 0.02739726 0.01369863 0.        ]
+     [0.03488372 0.90406977 0.05813953 0.00290698 0.        ]
+     [0.         0.05919003 0.84735202 0.09034268 0.00311526]
+     [0.         0.         0.05811623 0.92985972 0.01202405]
+     [0.         0.         0.         0.14285714 0.85714286]]
+    [[0.82692308 0.15384615 0.         0.01923077 0.        ]
+     [0.0703125  0.7890625  0.125      0.015625   0.        ]
+     [0.00295858 0.06213018 0.82248521 0.10946746 0.00295858]
+     [0.         0.00185529 0.07606679 0.88497217 0.03710575]
+     [0.         0.         0.         0.07803468 0.92196532]]
+    [[1.         0.         0.         0.         0.        ]
+     [0.         1.         0.         0.         0.        ]
+     [0.         0.06666667 0.9        0.03333333 0.        ]
+     [0.         0.         0.05660377 0.90566038 0.03773585]
+     [0.         0.         0.         0.03932584 0.96067416]]
+    >>> sm.S[0]
+    array([[0.54148249, 0.16780007, 0.24991499, 0.04080245, 0.        ],
+           [0.        , 0.        , 0.        , 0.        , 1.        ]])
+    >>> sm.S[2]
+    array([0.03607655, 0.22667277, 0.25883041, 0.43607249, 0.04234777])
+
```

Comment:
Given this discussion, shouldn't we always fill the empty states? i.e. shouldn't this always be `True`?

Reply:
Two reasons not filling empty states by default: 

* Right now, the tests for spatial dependence in the Markov chain model actually ignore the rows full of 0 estimates. If we fill empty states by default, it could have significant implications to these tests.
* Users might want to know where they have no observations.

Although the returned estimated transition probability matrix could have rows full of 0, when estimating the steady state distributions (for absorbing Markov chains) and first mean passage times, internally the transition probability matrix is adjusted so that it is stochastic - otherwise, it would not be possible to estimate them https://github.com/weikang9009/giddy/blob/limiting/giddy/markov.py#L226